Try to solve the problem of changing file name case from the zip file

drix00 opened the pull request:
This new version, lower case the file with extension ['.bas', '.frm', '.frx', '.vbp', '*.vbw'](Visual Basic file).
We have to do the first commit manually as most of the file will be new. After the first commit, git should be able to notice of file change even if the two files does not have the same case.
Example:
in git: ZAF3.bas
in zip with modification: zaf3.bas

> git status
> modified:   zaf3.bas

Hendrix

drix00 committed:
Add arguments to run the script locally.
Add function to lowercase files from visual basic.

drix00 commented:
The previous fix did not work because the option: `git config --global core.ignorecase true` work correctly on windows, but not on linux. On Linux, git still have two files when they don't have the same case and mark them as modified. I think we should remove it from travis configuration file.

Also maybe we should add check after the commit or before return statement when no change is reported to compare for file from the zip and in the working folder and make travis fail if it found a difference between the two files. So will know of future problem with the filename case.

Hendrix

drix00 commented on .travis/sync.py:
```diff
@@ -127,6 +127,28 @@ def extract(filepath, workdir):
         z.extractall(workdir)
     logger.info('Extracting %s... DONE', filepath)
 
+def all_files_lowercase(workdir):
```
Need to have a better method name.

drix00 commented on .travis/sync.py:
```diff
@@ -127,6 +127,28 @@ def extract(filepath, workdir):
+
+    ignore_paths = [".git", ".travis"]
+    ignore_files = [".gitignore", ".travis.yml", "VERSION.TXT"]
+    rename_file_extentions = ['.bas', '.frm', '.frx', '.vbp', '*.vbw']
```
Spelling mistake;(

drix00 commented on .travis/sync.py:
```diff
@@ -216,7 +248,9 @@ def main():
     url = args.url
     userzip = True
-    if urllib.parse.urlparse(url).scheme != '':
+    if urllib.parse.urlparse(url).scheme == 'd':
```
On windows, the scheme for a file path as url is 'd', but I did not find the documentation for that in the urllib doc. I did not check if it was the case on Linux, but this part of the code was working on Linux.

ppinard committed:
Merge branch 'filename_case' of github.com:openmicroanalysis/calczaf into filename_case

ppinard commented:
Is there a particular reason why to only change these file extensions ('.bas', '.frm', '.frx', '.vbp', '.vbw') to lower case and not all files?

drix00 commented:
No very good reason, just try to minimize the number of files to change. I also had a problem with "VERSION.TXT" when it was lower case for the rest of the script. I will test both solution and try to clean-up that part of the code.

ppinard committed:
Rename files to lowercase

ppinard committed:
Add script to rename files to lowercase

ppinard committed:
Rename all files in ZIP to lowercase. Renaming is performed as files
and directories in the ZIP are extracted.

ppinard committed:
Change the script so any change is committed even if there is no
entry in the VERSION.TXT. In this case, the commit message is
simply: Auto-sync [...]. In the other case, the text from VERSION.TXT
is used in the commit message.

drix00 committed:
Correct problem on windows when the url is a lcoal zip file.

drix00 commented:
To test the script locally on windows:

```
py -3 .travis\sync.py -v -w "C:\hdemers\temp\workFolder" -u "C:\hdemers\temp\zip\CALCZAF_SOURCE-E2.ZIP" --no-push --no-commit --reposurl "D:\work\codings\openmicroanalysis\calczaf" 
```

You may need to clean-up the workFolder and manually download the latest source zip file.

Example of output with the new ZIP file:

```
Commit disabled
Push disabled
Running git pull/clone...
Cloning into 'C:\hdemers\temp\workFolder'...
done.
Checking out files:  32% (114/352)   
...
Checking out files: 100% (352/352)   
Checking out files: 100% (352/352), done.
Running git pull/clone... DONE
Downloading...
Reading current version...
Reading current version... DONE
Reading zip version...
Reading zip version... DONE
Comparing versions...
Comparing versions... DONE
Extracting C:\hdemers\temp\zip\CALCZAF_SOURCE-E2.ZIP...
Extracting C:\hdemers\temp\zip\CALCZAF_SOURCE-E2.ZIP... DONE
Running git status...
Running git status... DONE
5 new changes found
Create commit message...
Create commit message... DONE
Message: Auto-sync on 2015/12/17 11:24:54

2015/12/06
    New PE graphics for peaking and PHA scanning.  Updated matrix.mdb file with 32 new actinides, 12 REE and Ti-Pt binaries.
2015/12/08
    Add index to MPB manual flag table (Poeml).
    Add Display current alpha factors menu for selected sample in Analyze!
2015/12/10
    New unk/std plot graphics for Output menu using Pro Essentials code.
2015/12/14
    Modify JEOL 8900/8200/8500 image shift set/get code to use micron units (Ma).
2015/12/15
    Modify TDI plotting in Standard Assignments to display nth points if desired.
Commit change.
Running git add...
Running git add... DONE
Running git commit...
On branch filename_case
Your branch is up-to-date with 'origin/filename_case'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

    modified:   addstd.frm
    modified:   analyze3.bas
    modified:   calczaf.vbp
    modified:   calczaf.vbw
    modified:   calczafplotalphas.bas
    modified:   clplot.bas
    modified:   edsplot.bas
    modified:   formula.frm
    modified:   global.bas
    modified:   init.bas
    modified:   init1.bas
    modified:   match.bas
    modified:   pe8api.bas
    modified:   penepma08plot.bas
    modified:   penepma12.bas
    modified:   penepma12b.bas
    modified:   penepma12matrix2.bas
    modified:   penepma12plot.bas
    modified:   standar2.bas
    modified:   update.bas
    modified:   version.txt
    modified:   zaf1.bas
    modified:   zaf3.bas
    modified:   zoom.bas

Running git commit... DONE
Running git push...
Everything up-to-date
Running git push... DONE
```

drix00 committed:
Correct problem on linux when the url is a local zip file.

drix00 commented:
To test locally on linux:

```
python3 .travis/sync.py -v -w "/home/hdemers/temp/workFolder" -u "/home/hdemers/temp/zip/CALCZAF_SOURCE-E2.ZIP" --no-push --no-commit --reposurl "/home/hdemers/work/codings/openmicroanalysis_github/calczaf"
```

Example of results with the sample zip file than the test on windows:

```
Commit disabled
Push disabled
Running git pull/clone...
Cloning into '/home/hdemers/temp/workFolder'...
done.
Running git pull/clone... DONE
Downloading...
Reading current version...
Reading current version... DONE
Reading zip version...
Reading zip version... DONE
Comparing versions...
Comparing versions... DONE
Extracting /home/hdemers/temp/zip/CALCZAF_SOURCE-E2.ZIP...
Extracting /home/hdemers/temp/zip/CALCZAF_SOURCE-E2.ZIP... DONE
Running git status...
Running git status... DONE
5 new changes found
Create commit message...
Create commit message... DONE
Message: Auto-sync on 2015/12/17 17:02:51

2015/12/06
    New PE graphics for peaking and PHA scanning.  Updated matrix.mdb file with 32 new actinides, 12 REE and Ti-Pt binaries.
2015/12/08
    Add index to MPB manual flag table (Poeml).
    Add Display current alpha factors menu for selected sample in Analyze!
2015/12/10
    New unk/std plot graphics for Output menu using Pro Essentials code.
2015/12/14
    Modify JEOL 8900/8200/8500 image shift set/get code to use micron units (Ma).
2015/12/15
    Modify TDI plotting in Standard Assignments to display nth points if desired.
Commit change.
Running git add...
Running git add... DONE
Running git commit...
On branch filename_case
Your branch is up-to-date with 'origin/filename_case'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

    modified:   addstd.frm
    modified:   analyze3.bas
    modified:   calczaf.vbp
    modified:   calczaf.vbw
    modified:   calczafplotalphas.bas
    modified:   clplot.bas
    modified:   edsplot.bas
    modified:   formula.frm
    modified:   global.bas
    modified:   init.bas
    modified:   init1.bas
    modified:   match.bas
    modified:   pe8api.bas
    modified:   penepma08plot.bas
    modified:   penepma12.bas
    modified:   penepma12b.bas
    modified:   penepma12matrix2.bas
    modified:   penepma12plot.bas
    modified:   standar2.bas
    modified:   update.bas
    modified:   version.txt
    modified:   zaf1.bas
    modified:   zaf3.bas
    modified:   zoom.bas

Running git commit... DONE
Running git push...
Everything up-to-date
Running git push... DONE
```

drix00 committed:
Change Travis configuration to checkout filename_case branch.

Commit:
Auto-sync on 2015/12/17 22:16:10

2015/12/06
    New PE graphics for peaking and PHA scanning.  Updated matrix.mdb file with 32 new actinides, 12 REE and Ti-Pt binaries.
2015/12/08
    Add index to MPB manual flag table (Poeml).
    Add Display current alpha factors menu for selected sample in Analyze!
2015/12/10
    New unk/std plot graphics for Output menu using Pro Essentials code.
2015/12/14
    Modify JEOL 8900/8200/8500 image shift set/get code to use micron units (Ma).
2015/12/15
    Modify TDI plotting in Standard Assignments to display nth points if desired.

drix00 commented:
I think the bug is resolved in this branch. I got the same modified file on windows and linux when I test manually (locally). Changing the branch to filename_case in Travis configuration file make the continuous integration to work.
I will wait the confirmation @ppinard  to merge with the master branch and close this branch. We have to remember to change back .travis.yml to the master branch.

ppinard commented:
I am happy to hear that it works on all platforms. Let's merge!

drix00 committed:
Merge branch 'filename_case'

drix00 committed:
Merge branch 'master' of https://github.com/openmicroanalysis/calczaf

drix00 committed:
Merge filename_case branch with master branch.
